pages/2_subscriptions_report.py

```diff
@@ -53,11 +133,141 @@
     with row1_col1:
         st.markdown("**Number of New Subscriptions**")
 
+        # Number of New Subscriptions
+        new_subscriptions_data = subscription_data.groupby('subscription_started_month').size()
+
+        # Convert PeriodIndex to timestamp for better x-axis formatting
+        new_subscriptions_data.index = new_subscriptions_data.index.to_timestamp()
+
+        # Convert Series to DataFrame
+        new_subscriptions_df = new_subscriptions_data.reset_index(name='count')
+        new_subscriptions_df.rename(columns={'index': 'subscription_started_month'}, inplace=True)
+
+        # Create a Plotly line chart
+        fig1 = px.bar(new_subscriptions_df, x='subscription_started_month', y='count')
+
+        # Set bar color
+        fig1.update_traces(marker_color='#306BEA')
+
+        # Update x and y labels
+        fig1.update_layout(
+            xaxis_title='',
+            yaxis_title=''
+        )
+
+        # Streamlit plot chart
+        st.plotly_chart(fig1)
+
     with row1_col2:
         st.markdown("**Number of New Subscriptions by Plan**")
+        
+        # Group by 'subscription_month' and 'subscription_plan'
+        subscription_by_plan = subscription_data.groupby(['subscription_started_month', 'subscription_plan']).size().unstack(fill_value=0)
+        
+        # Convert PeriodIndex to datetime for better x-axis formatting
+        subscription_by_plan.index = subscription_by_plan.index.to_timestamp()
+
+        # Convert DataFrame for plotting
+        subscription_by_plan_df = subscription_by_plan.reset_index()
+        subscription_by_plan_df = pd.melt(subscription_by_plan_df, id_vars=['subscription_started_month'], var_name='subscription_plan', value_name='count')
+
+        # Define custom colors for each subscription plan
+        color_map_2 = {
+            'Basic': '#306BEA',
+            'Premium': '#9EA91F',
+            'Standard': '#DB6645'
+        }
```

Comment:
This is great and makes for a VERY engaging chart. However, it is worth calling out the limitation here with hard coding the the subscription plans. If we were to expand this to allow others to connect their data to this Streamlit app then they would not have a very engaging chart since we have hard coded these values and do not have a dynamic mechanism to handle this. 

I would recommend we keep this as is for this build, but something to callout for the future to explore dynamic capabilities.

Reply:
I updated this to use a non-specific list.